openedx/core/djangoapps/user_api/accounts/serializers.py

```diff
@@ -48,6 +84,15 @@ def validate_name(self, attrs, source):
 
         return attrs
 
+    def validate_language_proficiencies(self, attrs, source):
+        """ Enforce all languages are unique. """
+        language_proficiencies = [language for language in attrs.get(source, [])]
+        if language_proficiencies and len(language_proficiencies) > 1:
+            first_code = language_proficiencies[0].code
```

Comment:
Aren't you just checking that none of the entries are equal to the first entry? What about checking uniqueness beyond the first entry?

Make sure you have tests for this method. I'm suspicious about whether it works even for just 2 entires (since the first entry will equal itself).